cdap-docs/reference-manual/source/http-restful-api/metadata.rst

```diff
@@ -0,0 +1,529 @@
+
+To delete a specific user metadata tag for an application, dataset, or stream, submit
+an HTTP DELETE request with the tag::
+
+  DELETE /v3/namespaces/<namespace>/<element-type>/<element-id>/metadata/properties/<tag>
+  
+or, for a particular user metadata tag of a program of a specific application::
+
+  DELETE /v3/namespaces/<namespace>/apps/<app-id>/<program-type>/<program-id>/metadata/properties/<tag>
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``<namespace>``
+     - Namespace ID
+   * - ``<element-type>``
+     - One of ``apps``, ``datasets``, or ``streams``
+   * - ``<element-id>``
+     - Name of the element
+   * - ``<app-id>``
+     - Name of the application
+   * - ``<program-type>``
+     - One of ``flows``, ``mapreduce``, ``spark``, ``workflows``, ``services``, or ``workers``
+   * - ``<program-id>``
+     - Name of the program
+   * - ``<tag>``
+     - Metadata tag
+
+.. rubric:: HTTP Responses
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Status Codes
+     - Description
+   * - ``200 OK``
+     - The method was successfully called, and the tags were deleted, or in the case of a
+       specific tag, was either deleted or the tag was not present
+   * - ``404 NOT FOUND``
+     - The entity (element or program) for which tags are being deleted was not found
+
+
+.. _http-restful-api-metadata-searching:
+
+Searching for Metadata
+======================
+To find which applications, datasets, or streams have a particular user metadata property or
+user metadata tag, submit an HTTP GET request::
+
+  GET /v3/namespaces/<namespace>/metadata/search?query=<term>&target=<element-type>
+
+Entities with the specified terms are returned as list of entity IDs::
+
+  ["entity1", "entity2"]
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``<namespace>``
+     - Namespace ID
+   * - ``<element-type>``
+     - One of ``app``, ``dataset``, ``program``, or ``stream``
+   * - ``<term>``
+     - Query term, as described below
+
+.. rubric:: HTTP Responses
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Status Codes
+     - Description
+   * - ``200 OK``
+     - Entities IDs of entities with the metadata properties specified were returned as a
+       list of strings in the body of the response
+   * - ``404 NOT FOUND``
+     - No entities matching the specified query were found
+
+.. rubric:: Query Terms
+
+CDAP supports prefix-based search of metadata properties and tags. Search for specific tags by using
+either a complete or partial tag name (with the remainder specified by an asterisk ``*``). 
+
+Search for properties by specifying one of:
+
+- a complete property key-value pair, separated by a colon, such as ``type:production``
+- a complete key with a partial value, such as ``type:prod*`` or ``type:*``
+- a complete or partial value, such as ``prod*``
```

Comment:
@henry-cask : Can you provide me with output and info on how to search for tags?